src/hooks/useSlippage.ts

```diff
@@ -5,9 +5,10 @@ import { useMemo } from 'react'
 import { InterfaceTrade } from 'state/routing/types'
 import { slippageAtom } from 'state/swap/settings'
 
-export function toPercent(maxSlippage: number | undefined): Percent | undefined {
+export function toPercent(maxSlippage: string | undefined): Percent | undefined {
```

Comment:
what necessitated this change? feels kinda unrelated to the rest of the pr

Reply:
This is required for integrations to have smooth input. Otherwise, things like '' or '1.' will change to a number (like 0 or 1) and give a bad user experience. It's a follow-up for maxSlippage becoming a string, which is done because it's input, so it needs to be treated like a string and not a number for better UX.